### PR TITLE
Sorted restaurant and food endpoints

### PR DESCRIPTION
**Pull request by DominykasSip**
Added optional parameters to existing endpoint so that user can fetch already sorted objects.

### CODE REVIEW

File: backend/WebApi/Controllers/RestaurantController.cs [hunk 1]
```diff
@@ -23,13 +25,43 @@ public RestaurantController(IRestaurantService restaurantService)
...
         /// <returns></returns>
         [HttpGet(Name = nameof(GetAll))]
         [ProducesResponseType((int)HttpStatusCode.OK, Type = typeof(IEnumerable<RestaurantDto>))]
-        public IActionResult GetAll()
+        [ProducesResponseType((int)HttpStatusCode.BadRequest)]
+        public IActionResult GetAll(string sortOrder = null, [FromQuery] Coords userCoordinates = null)
         {
+            try
+            {
+                NewFolder.InputValidator.ValidateRestaurantSortOrder(sortOrder, userCoordinates);
```
Comment (audrius-savickas) [inline backend/WebApi/Controllers/RestaurantController.cs:38]: NewFolder?

File: backend/WebApi/Helpers/InputValidator.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
+﻿using Domain.Entities;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace WebApi.NewFolder
```
Comment (audrius-savickas) [inline backend/WebApi/Helpers/InputValidator.cs:7]: I guess this should be `WebApi.Helpers`?

Comment (DominykasSip) [reply backend/WebApi/Helpers/InputValidator.cs:7]: I just auto implemented it I don't know why it did it like that

Comment (DominykasSip) [reply backend/WebApi/Helpers/InputValidator.cs:7]: Gonna check it out

**Review by audrius-savickas — APPROVED** [review]
Seems good to me! Or maybe we are missing other sorting options @JesusGonzalezA @DonatasPuc ?

File: backend/WebApi/Helpers/InputValidator.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+            return sortOrder switch
+            {
+                "dist" => true,
+                "dist_desc" => true,
+                "name" => true,
+                "name_desc" => true,
+                null => true,
+                _ => throw new System.ArgumentException("Invalid sort order"),
+            };
```
Comment (JesusGonzalezA) [inline backend/WebApi/Helpers/InputValidator.cs:26]: We could store the valid sorting types in an enum or an array, and then check if the type passed is contained there, for example using [this method](https://docs.microsoft.com/en-us/dotnet/api/system.enum.isdefined?view=net-5.0).
Why? If we want to add a new value to sort, we should change the validator class, although the logic has not changed. This way we just change the correct file just to update the list of correct inputs. It is not necessary to do it but just as a comment. Very good job!!

**Comment by JesusGonzalezA** [comment]
> Seems good to me! Or maybe we are missing other sorting options @JesusGonzalezA @DonatasPuc ?

I don't really miss other options

File: backend/WebApi/Helpers/InputValidator.cs (hunk 3 above)
Comment (JesusGonzalezA) [reply backend/WebApi/Helpers/InputValidator.cs:26]: I approve the changes, but I just left the comment!

**Review by JesusGonzalezA — APPROVED** [review]

**Review by DonatasPuc — COMMENTED** [review]
Logic to sort food items is duplicated in two places. Maybe we should have a separate class or method to do this? @audrius-savickas @JesusGonzalezA @DominykasSip

**Comment by audrius-savickas** [comment]
> Logic to sort food items is duplicated in two places. Maybe we should have a separate class or method to do this? @audrius-savickas @JesusGonzalezA @DominykasSip

that's true, it should most probably be moved into some sort of Sorting class

File: backend/WebApi/Helpers/InputValidator.cs (hunk 3 above)
Comment (DominykasSip) [reply backend/WebApi/Helpers/InputValidator.cs]: Made it use an Enum

**Review by audrius-savickas — APPROVED** [review]

File: backend/WebApi/Helpers/InputValidator.cs [hunk 4]
```diff
@@ -27,7 +27,7 @@ public static bool ValidateRestaurantSortOrder(string sortOrder, Coords userCoor
         {
             if (userCoordinates == null && (sortOrder == "dist" || sortOrder == "dist_desc"))
             {
-                throw new ArgumentNullException("Invalid user coordinates.");
+                throw new Exception("Invalid user coordinates.");
```
Comment (DonatasPuc) [inline backend/WebApi/Helpers/InputValidator.cs:30]: Why did this change to regular exception?

**Review by DonatasPuc — APPROVED** [review]

**Pull request merged**